pkg/cmd/get.go

```diff
@@ -142,9 +235,144 @@ func (o *GetOptions) Complete(args []string) error {
+	var k8sTrimmed bool
+	for _, resource := range resources {
+		if !o.passesTimestampFilter(resource) {
+			continue
+		}
+
+		resourceWithAvailability := &ResourceWithAvailability{
+			Resource:  resource,
+			InCluster: true,
+			Archived:  false,
+		}
+
+		// Insert in sorted order (newest first) using binary search
+		idx := o.findInsertionIndex(result, resource)
+
+		// Insert at the correct position
+		if idx < len(result) {
+			result = append(result, nil) // Add space
+			copy(result[idx+1:], result[idx:])
+			result[idx] = resourceWithAvailability
+		} else {
+			result = append(result, resourceWithAvailability)
+		}
+
+		// Trim to limit to save memory
+		if len(result) > o.Limit {
+			result = result[:o.Limit]
+			k8sTrimmed = true
+		}
+	}
+
+	return result, k8sTrimmed
+}
+
+// mergeKubeArchiveResourcesOptimized efficiently merges KubeArchive resources with existing sorted list
+func (o *GetOptions) mergeKubeArchiveResourcesOptimized(existing []*ResourceWithAvailability, kubearchiveResources []*unstructured.Unstructured) ([]*ResourceWithAvailability, bool, bool) {
+	// Create a map of existing resources by UID for deduplication
+	var k8sTrimmed, kubeArchiveTrimmed bool
+	existingUIDs := make(map[string]int)
+	for idx, res := range existing {
+		existingUIDs[string(res.Resource.GetUID())] = idx
+	}
+
+	result := make([]*ResourceWithAvailability, 0, len(existing)+len(kubearchiveResources))
+	result = append(result, existing...)
+
+	for processedIdx, resource := range kubearchiveResources {
+		uid := string(resource.GetUID())
+
+		var insertIdx int
+		if k8sIdx, exists := existingUIDs[uid]; exists {
+			// Resource exists in both - mark as archived too
+			existing[k8sIdx].Archived = true
+		} else {
+			// Resource only in archive
+			resourceWithAvailability := &ResourceWithAvailability{
+				Resource:  resource,
+				InCluster: false,
+				Archived:  true,
+			}
+
+			// Insert in sorted order using binary search
+			insertIdx = o.findInsertionIndex(result, resource)
+
+			// Insert at the correct position
+			if insertIdx < len(result) {
+				result = append(result, nil) // Add space
+				copy(result[insertIdx+1:], result[insertIdx:])
+				result[insertIdx] = resourceWithAvailability
+			} else {
+				result = append(result, resourceWithAvailability)
+			}
+		}
+
+		// Trim to limit to save memory and stop processing if we have enough
+		if len(result) > o.Limit {
+			result = result[:o.Limit]
+			if insertIdx < len(result) {
+				k8sTrimmed = true
+			} else {
+				kubeArchiveTrimmed = true
+			}
+		}
+		if processedIdx > o.Limit || insertIdx > o.Limit {
+			break
+		}
+	}
+
+	return result, k8sTrimmed, kubeArchiveTrimmed
+}
+
+// findInsertionIndex finds the correct index to insert a resource to maintain sorted order (newest first)
+func (o *GetOptions) findInsertionIndex(resources []*ResourceWithAvailability, newResource *unstructured.Unstructured) int {
```

Comment:
I think you have to handle nil here and make sure that nil items are always sorted to the end of the list.

Something similar is done for `keepLastWhen` [here](https://github.com/kubearchive/kubearchive/blob/main/cmd/vacuum/publisher.go#L289).

Reply:
but this is a private function only called in one place and nil isn't ever going to be a possible value. Either way I will handle it